### PR TITLE
kallisto version 0.42.1

### PR DESCRIPTION
**Pull request by pmelsted**
kallisto: requires c++11
kallisto 0.42.1

### CODE REVIEW

**Comment by iml** [comment]
Looks like `ENV.cxx11` is needed as well, at least for libstdc++.
```
fatal error: 'unordered_map' file not found
#include <unordered_map>
         ^
1 error generated.

**Comment by sjackman** [comment]
@iml Their Makefile already adds `-std=c++11`, so I'm not sure that `ENV.cxx11` would help here. See https://github.com/pachterlab/kallisto/blob/master/CMakeLists.txt#L8

I'm not sure that there's any combinations of options that will move Mountain Lion's `<tr1/unordered_map>` to `<unordered_map>`.

**Comment by saketkc** [comment]
Pass `-stdlib=libstdc++`  flag?

**Comment by sjackman** [comment]
It looks like the fix is

```ruby
    ENV.libcxx if MacOS.version < :mavericks && ENV.compiler == :clang
```

References:
https://github.com/Homebrew/homebrew/issues/25986#issuecomment-32705182
http://stackoverflow.com/questions/10438253/im-having-some-trouble-with-c11-in-xcode

**Comment by sjackman** [comment]
@pmelsted Can you add the above line to `kallisto` and re-push the pull request?
See also https://github.com/Homebrew/homebrew-science/blob/master/corset.rb

**Comment by sjackman** [comment]
@Homebrew/owners Should `needs :cxx11` on Mountain Lion also imply `ENV.libcxx` to provide `<unordered_map>`?

**Comment by tdsmith** [comment]
What Ian said; `needs` chooses compatible compilers and the `ENV` calls modify the superenv environment.

`ENV.cxx11` and `ENV.libcxx` enable different configurations in superenv shims:
https://github.com/Homebrew/homebrew/blob/master/Library/Homebrew/extend/ENV/super.rb#L291-L305

compiler | ENV.cxx11 | ENV.libcxx
--- | --- | ---
clang | x, g | g
gcc | x | (none)

`x` is c++11 mode, adding `-std=c++11`; `g` is libc++ mode, adding `-stdlib=libc++`.

libc++ is already the default on OS X >= Mavericks, so just writing `ENV.libcxx` is sufficient. If the code needs to be compiled in C++11 mode, just writing `ENV.cxx11` is sufficient.

**Comment by sjackman** [comment]
I've been schooled, Pall. Sorry for the backing and forthing. Can you please replace
`ENV.libcxx if MacOS.version < :mavericks && ENV.compiler == :clang`
with
`ENV.cxx11`
Thanks! This should be the last go around.

**Comment by sjackman** [comment]
The good news is Mountain Lion built this time! Yeah!

**Comment by pmelsted** [comment]
can this be merged now?

File: kallisto.rb [hunk 1]
```diff
@@ -11,10 +11,13 @@ class Kallisto < Formula
     sha256 "b8d02c19f47e73e56fcc976e957bcc22d008dae7f603652e329298d68c409074" => :mavericks
   end
 
+
```
Comment (sjackman) [inline kallisto.rb]: Please check the diff for unnecessary white space before submitting.

**Comment by sjackman** [comment]
Yes!

**Comment by sjackman** [comment]
Please squash new commits into one commit per formula. See [How To Open a Homebrew Pull Request (and get it merged)](https://github.com/Homebrew/homebrew/blob/master/share/doc/homebrew/How-To-Open-a-Homebrew-Pull-Request-(and-get-it-merged).md#how-to-open-a-homebrew-pull-request-and-get-it-merged)

**Comment by sjackman** [comment]
I've squashed the commits and fixed the white space.

**Comment by sjackman** [comment]
There was something wrong with the SHA256 calculated by `brew pull --bottle`. I fixed them up manually.

```diff
diff --git a/kallisto.rb b/kallisto.rb
index add9b3e..8ea2ab5 100644
--- a/kallisto.rb
+++ b/kallisto.rb
@@ -7,8 +7,8 @@ class Kallisto < Formula
   bottle do
     root_url "https://homebrew.bintray.com/bottles-science"
     cellar :any
-    sha256 "8acf76aabb96aeb0a5a87db13028e2c7c5d4c57058edada8dc1832c4ecb1221f" => :yosemite
-    sha256 "582c4fd9ae06d3c7a54bf45cdcc3599ad6da67875cf544f829831874462c1f19" => :mavericks
+    sha256 "4ac31c5cbe9ea2cbf9f707571b334c815e110c39af02fbd5377990873b71ac3b" => :yosemite
+    sha256 "396690ea206684d83dc21066d4bdbae74d00033d029ccad3634a9a415cbb0b9d" => :mavericks
     sha256 "4cdbabe76ca4741150919680b8d1418c0cdfe43876cd864c4b935447533223fc" => :mountain_lion
   end

 ```

**Comment by sjackman** [comment]
Merged. Thanks for your contribution, Pall.

**Pull request closed without merging**